library/core/src/iter/traits/iterator.rs

```diff
@@ -4018,4 +4018,66 @@ impl<I: Iterator + ?Sized> Iterator for &mut I {
     fn nth(&mut self, n: usize) -> Option<Self::Item> {
         (**self).nth(n)
     }
+    fn fold<B, F>(self, init: B, f: F) -> B
```

Comment:
Does this let us remove <https://stdrs.dev/nightly/x86_64-unknown-linux-gnu/core/iter/struct.ByRefSized.html>?

Reply:
Mostly, but the `#[inline]`s differ so that would affect performance and the tradeoffs aren't straight-forward.